### ISSUE TITLE
Compile of App fails when using customized Application App

### ISSUE THREAD

**Issue by ChrisBlankDe** [microsoft/navcontainerhelper#974]
**Describe the issue**
We try to compile an app which has a reference to `"application": "15.3.0.0"` in app.json.
The application.app was modified by us (own publisher and id).
Our customized application (and all of there dependencies to propagate) is located in the Symbols folder (`"Any Publisher_Application_15.3.0.0.app"`) but not published in the container.
There is no Application App published  in container cause its created with `-useNewDatabase`.

**Scripts used to compile and cause the issue**
```
Compile-AppInNavContainer -appProjectFolder C:\ProgramData\NavContainerHelper\Extensions\anycontainer\App -containerName anycontainer -credential $Credential -appSymbolsFolder "C:\ProgramData\NavContainerHelper\Extensions\anycontainer\Symbols"
```

**Full output of scripts**
```
Using Symbols Folder: C:\ProgramData\NavContainerHelper\Extensions\anycontainer\Symbols
Downloading symbols: Microsoft_Application_15.3.40074.41933.app
Url : http://172.29.186.93:7049/BC/dev/packages?publisher=Microsoft&appName=Application&versionText=15.3.0.0&tenant=default
Compile-AppInNavContainer : Ausnahme beim Aufrufen von "DownloadFile" mit 2 Argument(en):  "Der Remoteserver hat einen Fehler zurückgegeben: (404) 
Nicht gefunden."
In Zeile:1 Zeichen:1
+ Compile-AppInNavContainer -appProjectFolder C:\ProgramData\NavContain ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Compile-AppInNavContainer], MethodInvocationException
    + FullyQualifiedErrorId : WebException,Compile-AppInNavContainer
 
```

**Additional context**
I think the Problem is maybe around here: https://github.com/microsoft/navcontainerhelper/blob/5441d2b2e31307d99ec097b813f7321d361026e3/AppHandling/Compile-AppInNavContainer.ps1#L289
Currently publisher and name of the app will always be compared. The problem could be solved by not checking the publisher when the app is called "Application", right?

**Comment by freddydk** [comment]
I think you are right.
Have you tried to set Publisher to Microsoft (of the Application.app)? - then I would assume it works and the propagated dependencies should be loaded.

**Comment by ChrisBlankDe** [comment]
You're right. If I change Publisher of Application App to "Microsoft" everything works fine.
But it should also work with a custom Publisher, isn't it?

**Comment by freddydk** [comment]
It should. but in the meeting yesterday, I heard for the first time that the recommendation was to change the publisher of the Application App as well:-(
I will fix the containerhelper, but in the meantime, you can keep the publisher as Microsoft for the Application App only.

**Comment by ChrisBlankDe** [comment]
that's what I'm going to do. I'm just asked again to make sure we got it right. :)
THX

**Comment by freddydk** [comment]
We will keep this issue open so you can see when this is fixed in NavContainerHelper

**Comment by freddydk** [comment]
Shipped in 0.6.5.9
